x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/support/UserRoleMapper.java

```diff
@@ -222,31 +276,34 @@ public boolean test(FieldExpression.FieldValue fieldValue) {
                     final String suffix = pattern.substring(2);
                     // if the suffix has a wildcard, then it's not a pure sub-tree match
                     if (suffix.indexOf('*') == -1) {
-                        final DN dnSuffix = parseDn(suffix);
-                        if (dnSuffix != null && dn.isDescendantOf(dnSuffix, false)) {
-                            return true;
-                        }
+                        return isDescendantOf(dnNormalizer.normalize(suffix));
```

Comment:
The comment above (`// We can use dn.isDescendantOf for that`) is now out of date with the implementation.

Reply:
Removed. Thanks!